### PR TITLE
Complete av1d output surface information.

### LINKED ISSUE #2231: AV1d doesn't fill required surface fields
timestamp, aspect ratio, FrameRateExt, FrameOrder, DataFlag, mfxExtDecodedFrameInfo, etc handling are missed in #2218 _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp

### PR DESCRIPTION
**Pull request by wangyan42164**
Fixes #2231.

Signed-off-by: Yan Wang <yan.wang@linux.intel.com>

### CODE REVIEW

File: _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp [hunk 1]
```diff
@@ -1171,8 +1190,14 @@ mfxStatus VideoDECODEAV1::FillOutputSurface(mfxFrameSurface1** surf_out, mfxFram
     UMC::VideoDataInfo const* vi = fd->GetInfo();
     MFX_CHECK(vi, MFX_ERR_DEVICE_FAILED);
 
+    surface_out->Data.TimeStamp = bs->TimeStamp != static_cast<mfxU64>(MFX_TIMESTAMP_UNKNOWN) ? bs->TimeStamp : GetMfxTimeStamp(m_frame_order * m_in_framerate);
```
Comment (IlyaSergeyev) [inline _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp:1193]: Seems using bitstream here is not good idea. It is better to store timestamp in the pFrame struct (somewhere in the umc_av1_* code) and then just use GetMfxTimeStamp() function here.

Comment (dmitryermilov) [reply _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp:1193]: Echo Ilya. It's better to set/get TS via pFrame . Two reasons:
1) Just to code style (we do this way for the rest decoders)
2) I assume AV1 has a re-ordering. If so, TS from bistream (in encoded order) can't be set to decoded surface (in display order).

File: _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp [hunk 2]
```diff
@@ -1171,8 +1190,14 @@ mfxStatus VideoDECODEAV1::FillOutputSurface(mfxFrameSurface1** surf_out, mfxFram
...
+    surface_out->Data.FrameOrder = m_frame_order;
+    surface_out->Data.Corrupted = 0;
+    surface_out->Info.PicStruct = MFX_PICSTRUCT_PROGRESSIVE;
     surface_out->Info.CropW = static_cast<mfxU16>(pFrame->GetUpscaledWidth());
-    surface_out->Info.CropH = static_cast<mfxU16>(pFrame->GetHeight());
+    surface_out->Info.CropH = static_cast<mfxU16>(pFrame->GetFrameHeight());
+    surface_out->Info.AspectRatioW = 1;
+    surface_out->Info.AspectRatioH = 1;
```
Comment (dmitryermilov) [inline _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp:1200]: There is still no:
- FrameRate info.
- mfxExtDecodedFrameInfo
 - proper handling of  Corrupted

Comment (wangyan42164) [reply _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp:1200]: FrameRateInfo: I have updated in the latest commit.
mfxExtDecodedFrameInfo: I think we have done it in SetFrameType() which is called by :
https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp#L126-L146
https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp#L1167
Corrupted: IMHO, AV1 should have not special handling like VP9. @IlyaSergeyev Do you think so?

Comment (dmitryermilov) [reply _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp:1200]: Okay. Thanks. So only 'Corrupted' remained not covered here.

Comment (dmitryermilov) [reply _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp:1200]: Ahh, I just saw it's already covered here: https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp#L674

File: _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp [hunk 3]
```diff
@@ -1149,10 +1162,16 @@ inline void CopyFilmGrainParam(mfxExtAV1FilmGrainParam &extBuf, UMC_AV1_DECODER:
...
+mfxStatus VideoDECODEAV1::FillOutputSurface(mfxBitstream *bs, mfxFrameSurface1** surf_out, mfxFrameSurface1* surface_work, UMC_AV1_DECODER::AV1DecoderFrame* pFrame)
 {
     MFX_CHECK_NULL_PTR3(surface_work, pFrame, surf_out);
 
+    if (!pFrame->GetShowFrame())
+    {
+        surf_out = 0;
+        return (mfxStatus)MFX_ERR_MORE_DATA_SUBMIT_TASK;
+    }
+
```
Comment (dmitryermilov) [inline _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp:1174]: It should be handled in GetFrameToDisplay(), not in the FillOutputSurface function which just adds some meta info to mfxFrameSurface1

Comment (wangyan42164) [reply _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp]: Yes. Its has been checked in the following:
https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/shared/umc/codec/av1_dec/src/umc_av1_decoder.cpp#L546
I have removed the section in the latest commit.
Thanks.

File: _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp (hunk 1 above)
Comment (dmitryermilov) [reply _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp:1187]: @wangyan42164 , everything except this part is fine to me. Can you please re-factor the item which Ilya pointed at? I understand that you borrowed the logic from VP9 but it wan't written a good way there.

Comment (wangyan42164) [reply _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp]: Sure. I have changed it in the latest commit. @dmitryermilov @dmitry-gurulev @IlyaSergeyev Could you please review it again? Thanks.

**Comment by wangyan42164** [comment]
@dmitryermilov I have refined this PR based on our offline discussion. Could you please review it again? Thanks.

**Pull request merged**